async/kafka/option.go

```diff
@@ -53,3 +54,11 @@ func Start(offset int64) OptionFunc {
 		return nil
 	}
 }
+
+// Decoder option for injecting a specific decoder implementation
+func Decoder(dec encoding.DecodeRawFunc) OptionFunc {
+	return func(c *consumer) error {
+		c.dec = dec
```

Comment:
We should check if the decoder is nil and return an error and add the appropriate tests.

Reply:
we are making the nil check later anyway ...
happy to add the nil-check , but it would not avoid any breaking case,
just prompt the user in case they mess something up.

Reply:
I agree that we check it later, but the user intends to set up a decoder and we should warn him if the decoder is nil.
If we don't the code will actually try the content type in order to find out the decoder.